docs/proposals/multiple-schedulers.md

```diff
@@ -0,0 +1,139 @@
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+<strong>
+The latest 1.0.x release of this document can be found
+[here](http://releases.k8s.io/release-1.0/docs/proposals/multiple-schedulers.md).
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# WIP: Multi-Scheduler in Kubernetes
+
+**Author**: Haiyang DING(@HaiyangDING)
+
+**Status**: Design & Implementation in progress.
+
+## Motivation
+
+In current Kubernetes design, there is only one default scheduler in a Kubernetes cluster.
+However in some scenarios, users may want to launch user-provided schedulers to enable some
+customized scheduling behavior. This is one of the reasons we should consider multiple schedulers
+in Kubernetes. As previously discussed in [#11793](https://github.com/kubernetes/kubernetes/issues/11793),
+[#9920](https://github.com/kubernetes/kubernetes/issues/9920) and
+[#11470](https://github.com/kubernetes/kubernetes/issues/11470), the design of the multiple scheduler should be generic.
+
+## Challenges in multiple schedulers
+
+- Separating the pods
+
+    Each pod should be scheduled by only one scheduler. As for implementation, a pod should
+    have an additional field to tell by which scheduler it wants to be scheduled. Besides,
+    each scheduler, including the default one, should have a unique logic of how to add unscheduled
+    pods to its to-be-scheduled pod queue. Details will be explained in later sections.
+
+- Dealing with conflicts
+
+    Different schedulers are essentially separated processes on the system level. When all schedulers
+    try to schedule their pods on the nodes, there might be conflicts.
+
+    One example of the conflicts is resource racing: Suppose there be a `pod1` scheduled by
+    `my-scheduler` requiring 1 CPU's *request*, and a `pod2` scheduled by `kube-scheduler` (k8s native
+    scheduler, acting as default scheduler) requiring 2 CPU's *request*, while `node-a` only has 2.5
+    free CPU's, if both schedulers all try to put their pods on `node-a`, then one of them would eventually
+    fail when kubelet on `node-a` performs the create action due to insufficient CPU resources.
+
+    This conflict is complex to deal with in api-server and etcd. Our current solution is to let kubelet
+    to do the conflict check and if the conflict happens, effected pods would be put back to scheduler
+    and waiting to be scheduled again. Implementation details are in later sections.
+
```

Comment:
I think most of these are either already sufficiently covered in the doc or don't need to be covered in this doc. The two I agree with are:

1) "How to plugin" - it could be useful to add a sentence to the Motivation section mentioning that we envision a number of ways to extend the scheduler: write your own scheduler (described in this doc), use the callout approach implemented in #13580, recompile the scheduler with the new policy (and select it using the config file), and maybe in the future dynamically link a new policy into the scheduler.

2) Mention explicitly in the introduction that we're not trying to address the question of how the scheduler name annotation gets set. (Though I think mentioning that we anticipate it will be set by a component like admission controller/initializer, as the doc currently does, is reasonable.)